core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -328,7 +358,7 @@ private static ServiceRequestContext serviceRequestContext() {
     }
 
     @Override
-    public CompletableFuture<Boolean> init(EndpointGroup endpointGroup) {
+    public CompletableFuture<Boolean> init() {
         assert endpoint == null : endpoint;
         assert !initialized;
         initialized = true;
```

Comment:
The context customer executed below was composed with a thread-local context customizer so `runContextCustomizer()` should be invoked in the caller thread.
If `HttpPreprocessor` changes the current thread, the thread-local context customizer won't work.

Do you think we need to change where we call `runContextCustomizer()?

https://github.com/line/armeria/blob/244e5cb661c8990a771db1696aa7da558fd7be4a/core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java#L276-L284

Reply:
I see, I missed that the contract is to run the customizer from the calling thread.

https://github.com/line/armeria/blob/1157370431082d25da20884b3c067237658028bf/core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java#L449-L450

Done at https://github.com/line/armeria/pull/6057/commits/b6b4fae1b4d11435fa2eae6f1308ebca1d0e3f7d